src/intelli/node/stringselection.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ * GTlab IntelliGraph
+ *
+ *  SPDX-License-Identifier: BSD-3-Clause
+ *  SPDX-FileCopyrightText: 2024 German Aerospace Center
+ *
+ *  Author: Jens Schmeink <jens.schmeink@dlr.de>
+ */
+#include "intelli/node/stringselection.h"
+
+#include "intelli/data/string.h"
+#include "intelli/data/stringlist.h"
+
+#include <QComboBox>
+
+using namespace intelli;
+
+StringSelectionNode::StringSelectionNode() :
+    Node(tr("String Selection")),
+    m_selection("selectedString", "selection", "selection")
+{
+    m_in = addInPort({typeId<StringListData>(), tr("List")});
+    m_out = addOutPort(makePort(typeId<StringData>()).setCaption(tr("Selection")));
+
+    registerProperty(m_selection);
+    m_selection.hide();
+
+    registerWidgetFactory([this]() {
+        auto w = std::make_unique<QComboBox>();
+
+        QStringList given;
+
+        if (auto list = nodeData<StringListData>(m_in))
+        {
+            given = list->value();
+        }
+
+        w->addItems(given);
+
+        if (given.isEmpty())
+        {
+            m_selection.setVal("");
+        }
+        else m_selection.setVal(given.first());
```

Comment:
I think this it would be more concise to use a ternary operator here
```suggestion
        m_selection = given.isEmpty() ? QString{} : given.first();
```